pandas/_libs/tslibs/period.pyx

```diff
@@ -1509,11 +1512,16 @@ cdef class _Period:
 
     cdef readonly:
         int64_t ordinal
+        PeriodDtype _dtype
         object freq
 
     def __cinit__(self, ordinal, freq):
         self.ordinal = ordinal
         self.freq = freq
+        # Note: this is more performant than PeriodDtype.from_date_offset(freq)
+        #  because from_date_offset cannot be made a cdef method (until cython
+        #  supported cdef classmethods)
+        self._dtype = PeriodDtype(freq._period_dtype_code)
```

Comment:
Since you basically only use this as `self._dtype.dtype_code` (from a quick look at the diff), why not storing the `dtype_code` itself instead of wrapped in a PeriodDtype?

(btw, the PeriodDtype is also not a dtype, so we need to find another name for it, I would say)

Reply:
> why not storing the dtype_code itself instead of wrapped in a PeriodDtype?

We could do this.  My expectation is that it will get more attributes before long.

Reply:
What other attributes are you thinking of?

Reply:
we frequently split this integer code up into two pieces, corresponding to

```
def decompose(code):
    head, anchor = divmod(code, 1000)
    grp = head * 1000
    return grp, anchor
```

im looking at storing grp and anchor instead of just the code.  The upside of this is that `grp` corresponds to the `FreqGroup` class in libfrequencies, so we could re-use that.

Reply:
OK, I don't mind too much. If let it to you to judge whether it is already worth adding the full class instead of just the `dtype_code` attribute. 

But, when keeping it:

- The other methods of the PeriodDtype class don't seem used and/or tested?
- It needs another name IMO, since we already have another PeriodDtype class that is something else